geode-core/src/main/java/org/apache/geode/internal/cache/TXManagerImpl.java

```diff
@@ -670,6 +669,9 @@ private void closeListener(TransactionListener tl) {
     }
   }
 
+  private final ConcurrentMap<Thread, Boolean> transactionInternalSuspendedByThreads =
```

Comment:
I think what you want to know is: "has the current thread internally suspended the tx"? I think to answer that question you should use a boolean ThreadLocal. You are kind of implementing your own thread local in this case by adding and remove from a collection the Thread itself. But then you also need to worry about cleaning up this collection if the thread dies without removing itself.
It would be ideal if we could find a way to do this that did not depend on a ThreadLocal but I don't think reimplementing ThreadLocal using a collection is an improvement

Reply:
Removed this by adding a PausedTXState so that product could detect if the same thread has paused transaction.